docs/modules/ROOT/pages/references/architecture/control-api-design.adoc

```diff
@@ -0,0 +1,181 @@
+= APPUiO Control API Design
+
+This page describes the overall APPUiO Control API ("API" in the rest of this page) design.
+It adheres to the decision taken in xref:explanation/decisions/control-api.adoc[APPUiO Control API].
+The design follows the features asked in xref:explanation/system/details-ui.adoc[System idea: Graphical user interface].
+
+image:reference/control-api-design.drawio.svg[]
+
+== General Principles
+
+Kubernetes API::
+The Control API is built upon the Kubernetes API and adheres to it's https://kubernetes.io/docs/reference/kubernetes-api/[design principles].
+
+API object field: Status::
+The `status` field (also called https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource["Status subresource"]) represents the current status and external information.
+It is read-only to the end-user.
+
+API object field: Spec::
+The `spec` field contains the desired state for reconciliation.
+
+=== Virtual vs. CRD based resources
+
+Some resources not persisted to etcd and are only available virtually (API extension), others are persisted to etcd and are defined and represented via `CustomResourceDefinition`s (CRDs).
+
+Virtual resources are accessible via an https://kubernetes.io/docs/tasks/extend-kubernetes/setup-extension-api-server/[API Server Extension].
+These resources are similar to views in a relational database.
+The benefit of providing these resources instead of only using CRDs is that we can calculate access permissions dynamically for every request.
+The same concept is also used by OpenShift with its `Project` resource which represents RBAC filter `Namespaces` (see https://github.com/openshift/kube-projects[kube-projects]).
+
+=== Authentication and Authorization
+
+Authentication against the API server is done by the {idp} by leveraging OIDC.
+This allows to transport authorized information about the user in a JWT token.
+
+For authorization, standard https://kubernetes.io/docs/reference/access-authn-authz/rbac/[Kubernetes RBAC] is being used.
+Kyverno policies can be used to implement enhanced policies, for example the number of resources of a specified kind a user is allowed to create.
```

Comment:
What this page doesn't cover is how the first access after registration works. Openshift first has to know the user before being able to authorize users or before Rolebindings take effect. Do newly registered users have to first login to the console of $someCluster (so that the user object gets created) before being able to use the API (and by extension, the Web UI)?

Reply:
The API doesn't have to run on OpenShift. I expect this API to run on a lightweight Kubernetes like vcluster.